Core/include/Acts/Seeding/SpacePointGrid.ipp

```diff
@@ -126,13 +122,120 @@ Acts::SpacePointGridCreator::createGrid(
 
   } else {
     // Use the zBinEdges defined in the config
+    zValues.reserve(config.zBinEdges.size());
     for (float bin : config.zBinEdges) {
       zValues.push_back(bin);
     }
   }
 
   detail::Axis<detail::AxisType::Variable, detail::AxisBoundaryType::Bound>
       zAxis(std::move(zValues));
-  return std::make_unique<Acts::SpacePointGrid<SpacePoint>>(
+  return Acts::SpacePointGrid<SpacePoint>(
       std::make_tuple(std::move(phiAxis), std::move(zAxis)));
 }
+
+template <typename external_spacepoint_t,
+          typename external_spacepoint_iterator_t, typename callable_t>
+void Acts::SpacePointGridCreator::fillGrid(
```

Comment:
Is this not effectively a method on `grid`? Or is there a reason this is better implemented as an external static method?

Reply:
In my opinion `fillGrid` should not be a method of the grid. The grid itself should know nothing about that, since that is an object that defines some axes.

The `fillGrid` methods requires the knowledge of what axis corresponds to what variable and also how to retrieve/compute those variables given (e.g.) a space points. I don't think the grid should care about this. It is the duty of the costumer to define a function for filling their own grid.

So in my mind one will have to create thei own CustomSpacePointGridCreator class and define there the way their grid is defined and the instructions on how to fill it.

Reply:
Meaning `SpacePointGridCreator` is supposed be just one way to use this grid type?